### ISSUE TITLE
Missing error for enum using non-number variables named `Infinity`/`NaN`

### ISSUE THREAD

**Issue by nicolo-ribaudo** [microsoft/TypeScript#54981]
# Bug Report

### 🔎 Search Terms

enum number infinity nan

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/N4KABGA2CmAuYEkB2AzAlktsCeYC8YwAvgNzhjRICuAtmAKJKHkQAa+iqGW25RI-EKAgx4AZQD2NOAAsMAc3qQAztA7EyESrQZNhEMOwKTpsOUkUrofASCA)

### 💻 Code

```ts
{
  let Infinity = {};
  enum En {
    X = Infinity
  }
}

{
  let NaN = {};
  enum En {
    X = NaN
  }
}

{
  let SomethingElse = {};
  enum En {
    X = SomethingElse
  }
}
```

### 🙁 Actual behavior

TS reports _"Type '{}' is not assignable to type 'number' as required for computed enum member values"_ only for `X = SomethingElse`

### 🙂 Expected behavior

TS should also report an error for `X = Infinity` and `X = NaN`.

**Comment by Andarist** [comment]
First things first - should `let Infinity = {}` be even considered legal? 😅

**Comment by Andarist** [comment]
Yeah, I know about those things - but since it's not-configurable you immediately get an error if u try to do it at runtime. So perhaps this should be recognized by TS as a problem? 🤔

**Comment by fatcerberus** [comment]
There's no runtime error for

```ts
let undefined = 777;
console.log(undefined);
```

Likewise for `Infinity`.  You only get a runtime error if you try to assign to the global binding, but shadowing it is fine.

**Comment by Andarist** [comment]
Ah, got it - got an error yesterday when I executed it in the console without adding extra scope/block 👍

**Comment by nicolo-ribaudo** [comment]
Also, in module code you do not need to add the extra block. Only in scripts.
